sdk/nodejs/dagger/dist/engine.js

```diff
@@ -38,9 +38,24 @@ export class Engine {
             this.config.Port = this.config.Port || 8080;
             args.push("--port", `${this.config.Port}`);
             const serverProc = execa("dagger", args, {
-                stdio: "inherit",
+                stdio: "pipe",
                 cwd: this.config.Workdir,
             });
+            try {
+                yield serverProc;
```

Comment:
In the happy-path case I think this results in just blocking on waiting for the subprocess, which itself just sits there waiting for input, so nothing happens (just tried locally quick).

Maybe instead we can wrap all of the rest of the code in the `try`? Or something similar? Or if that doesn't work we can fallback to parsing `version` output I suppose.

More fundamental issue here is that there's no automated tests for any of this yet so it's really hard to be aware of any of this, made an issue for that (totally separate from your effort here though, not your responsibility): https://github.com/dagger/dagger/issues/3290

Reply:
I used the todoapp example: https://github.com/dagger/todoapp/tree/cloak

Updated `package.json` there to use:
```
  "devDependencies": {
    "@dagger.io/dagger": "git+https://github.com/gerhard/dagger.git#guard-against-wrong-dagger-in-typescript-sdk"
  },
```

and ran `yarn run kick-the-tires`.

Reply:
I'm gonna see if it takes like 10 mins to add an automated happy-path test for the engine. If it does I'll push and update you so there's less silly hoops to jump through for all this.

Reply:
PR for that test here: https://github.com/dagger/dagger/pull/3292